### PR TITLE
Add a supporter promo to changelogs

### LINKED ISSUE #2367: Changelog should have a support promo
```
{if !$isSupporter}
<h1>Love this update?</h1>

<div class='paddingboth'>
	<big>Support further development of osu! and <strong><a href='/p/support'>become a supporter</a></strong> today! Not only will you help speed development, but you will also get some extra features and customisations!</big>
</div>
{/if}
```

something similar to this.

### PR DESCRIPTION
**Pull request by LiquidPL**
Fixes #2367.

### CODE REVIEW

File: resources/views/changelog/show.blade.php [hunk 1]
```diff
@@ -64,6 +64,11 @@ class="changelog__build-link"
                     @each('changelog._changelog_change', $changelogs, 'log')
                 </div>
 
+                @if(!Auth::check() || !Auth::user()->isSupporter())
```
Comment (LiquidPL) [inline resources/views/changelog/show.blade.php]: not sure if this should be only users without supporters, or also logged out ones, but that's how it is right now on the old website (i think)

Comment (jacksonisiah) [reply resources/views/changelog/show.blade.php]: currently this is how the old site behaves.

**Comment by nanaya** [comment]
There's a specific design for this somewhere.

**Comment by LiquidPL** [comment]
@peppy @arflyte can i have it then?

File: resources/views/changelog/show.blade.php (hunk 1 above)
Comment (peppy) [reply resources/views/changelog/show.blade.php]: I'm not sure you even need the first check here?

**Comment by LiquidPL** [comment]
how about that image alone in @2x? or perhaps the whole psd?

**Comment by arflyte** [comment]
[osu!next - Change Logs B.zip](https://github.com/ppy/osu-web/files/1730531/osu.next.-.Change.Logs.B.zip)

File: resources/views/changelog/show.blade.php (hunk 1 above)
Comment (LiquidPL) [reply resources/views/changelog/show.blade.php]: it will fail for logged out users otherwise.

**Comment by nekodex** [comment]
I think that image is already in use for supporter promo elsewhere, may want to check before re-including it.

**Comment by peppy** [comment]
Pop up when downloading 999 maps in a row i think

**Comment by LiquidPL** [comment]
There's another pippi there, however this one is used on the frontpage, without the heart tho:

image

**Comment by LiquidPL** [comment]
I think it ended up good 🤔 

image

File: resources/assets/less/bem/changelog.less [hunk 2]
```diff
@@ -17,12 +17,7 @@
  */
 
 .changelog {
-  padding: 20px;
-
-  @media @desktop {
-    padding-left: @padding-big;
-    padding-right: @padding-big;
-  }
+  padding: 20px 0;
```
Comment (nanaya) [inline resources/assets/less/bem/changelog.less]: this breaks padding on comment area

File: resources/assets/less/bem/supporter-promo.less [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    a {
+      color: @pink-dark;
+    }
+
+    &--small {
+      font-size: @font-size--normal;
+    }
+
+    &:first-of-type {
```
Comment (nanaya) [inline resources/assets/less/bem/supporter-promo.less]: why so specific

File: resources/assets/less/bem/supporter-promo.less [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    flex-direction: column;
+    align-items: center;
+
+    .at2x-simple('/images/layout/pippi.png');
+    background-size: 164px;
+
+    min-width: 164px;
+    height: auto;
+
+    margin-top: -28px;
```
Comment (nanaya) [inline resources/assets/less/bem/supporter-promo.less]: adjust using `background-position`?

File: resources/assets/less/bem/supporter-promo.less [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+.supporter-promo {
+  display: flex;
+  justify-content: space-between;
+
+  padding-left: 65px;
+  padding-right: 75px;
+  height: 200px;
```
Comment (nanaya) [inline resources/assets/less/bem/supporter-promo.less]: mobile should look more or less like this

File: resources/views/changelog/show.blade.php [hunk 6]
```diff
@@ -64,6 +64,22 @@ class="changelog__build-link"
...
+                    <div class="changelog__supporter-promo supporter-promo">
+                        <div class="supporter-promo__text-box">
+                            <p class="supporter-promo__heading">{{ trans('changelog.support.heading') }}
+
+                            <div>
+                                <div class="supporter-promo__text">{!! trans('changelog.support.text-1', ['url' => route('support-the-game')]) !!}</div>
```
Comment (nanaya) [inline resources/views/changelog/show.blade.php]: `p`

File: resources/views/changelog/show.blade.php [hunk 7]
```diff
@@ -64,6 +64,22 @@ class="changelog__build-link"
                     @each('changelog._changelog_change', $changelogs, 'log')
                 </div>
 
+                @if(!Auth::check() || !Auth::user()->isSupporter())
+                    <div class="changelog__supporter-promo supporter-promo">
+                        <div class="supporter-promo__text-box">
+                            <p class="supporter-promo__heading">{{ trans('changelog.support.heading') }}
```
Comment (nanaya) [inline resources/views/changelog/show.blade.php]: Some heading tag (`h2`).

File: resources/views/changelog/show.blade.php [hunk 8]
```diff
@@ -64,6 +64,22 @@ class="changelog__build-link"
                     @each('changelog._changelog_change', $changelogs, 'log')
                 </div>
 
+                @if(!Auth::check() || !Auth::user()->isSupporter())
+                    <div class="changelog__supporter-promo supporter-promo">
```
Comment (nanaya) [inline resources/views/changelog/show.blade.php]: split block and element.

File: resources/assets/less/bem/supporter-promo.less [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  background-color: #fff4f9;
```
Comment (nanaya) [inline resources/assets/less/bem/supporter-promo.less]: 75% pink-lighter

Comment (LiquidPL) [reply resources/assets/less/bem/supporter-promo.less]: how exactly? using `fade()`?

File: resources/assets/less/bem/changelog.less [hunk 10]
```diff
@@ -31,6 +26,26 @@
...
+  }
+
+  &__supporter-promo {
+    margin: 0 10px;
+
+    @media @desktop {
+      margin: 0 55px;
+    }
+
+    margin-top: 80px;
```
Comment (nanaya) [inline resources/assets/less/bem/changelog.less]: Not needed (already done in `__list`).

File: resources/assets/less/bem/changelog.less [hunk 11]
```diff
@@ -31,6 +26,26 @@
...
+  }
+
+  &__disqus {
+    @media @desktop {
+      padding: 0 @padding-big;
+    }
+  }
+
+  &__supporter-promo {
+    margin: 0 10px;
```
Comment (nanaya) [inline resources/assets/less/bem/changelog.less]: should have 20px margin bottom.

File: resources/lang/en/changelog.php [hunk 12]
```diff
@@ -28,4 +28,10 @@
         'fix' => 'fix',
         'misc' => 'miscellanous',
     ],
+
+    'support' => [
+        'heading' => 'Love this update?',
+        'text-1' => 'Support further development of osu! and <strong><a href=":url">become a supporter</a></strong> today!',
```
[the remark this answers is not in the document]
Comment (nanaya) [reply resources/lang/en/changelog.php]: And might as well attach a class to the `a` and set the styling there.

File: resources/lang/en/changelog.php [hunk 13]
```diff
@@ -28,4 +28,12 @@
         'fix' => 'fix',
         'misc' => 'miscellanous',
     ],
+
+    'support' => [
+        'heading' => 'Love this update?',
+        'text_1' => 'Support further development of osu! and',
+        'text_1_link' => 'become a supporter',
+        'text_1_end' => 'today!',
```
Comment (nanaya) [inline resources/lang/en/changelog.php]: ಠ_ಠ no

**Pull request merged**